cmd/fs-v1-helpers.go

```diff
@@ -365,6 +365,7 @@ func fsRenameFile(sourcePath, destPath string) error {
 
 // Delete a file and its parent if it is empty at the destination path.
 // this function additionally protects the basePath from being deleted.
+// This will return nil if called on a non-empty directory.
```

Comment:
Yes, this isn't actually a change in behavior. I added this comment to clarify what the function does (hence the confusion 😄 ), as [before](https://github.com/minio/minio/blob/ec5293ce29ef03726c7174d9e3304bc156259c4e/cmd/fs-v1-helpers.go#L392-L395) it would `return nil` from the `Stat()` call. This actually led to an interesting race condition (never became an issue afaik), because if `Stat()` returned a non-empty directory, but was empty by the time the `Remove()` call came around, [it would error](https://github.com/minio/minio/blob/ec5293ce29ef03726c7174d9e3304bc156259c4e/cmd/fs-v1-helpers.go#L404). So technically this PR fixes that race condition, even though I doubt it would ever come to fruition in a production environment.

Also, all tests that are written for this commit pass both before and after the changes. This was done to prevent any change in behavior.